From 591962db3e2a67f3b2d068a925698e939f4d37b7 Mon Sep 17 00:00:00 2001
From: Niklaus Xiao <stone_xy@live.cn>
Date: Tue, 16 May 2017 08:36:05 +0200
Subject: [PATCH 1118/1164] CDH-53942 HIVE-16413: Create table as select does
 not check ownership of the location (Niklaus Xiao
 via Zoltan Haindrich)

Signed-off-by: Zoltan Haindrich <kirk@rxd.hu>
(cherry picked from commit 91948ec00b1b642b401bfb529f211eb59b8b16ad)

Change-Id: I1711f836a24a7e7f32044d41b6760d88ace448d3
---
 .../plugin/sqlstd/Operation2Privilege.java         |    1 +
 1 file changed, 1 insertion(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/sqlstd/Operation2Privilege.java b/ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/sqlstd/Operation2Privilege.java
index 5d2eec5..42e50ff 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/sqlstd/Operation2Privilege.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/sqlstd/Operation2Privilege.java
@@ -281,6 +281,7 @@ public HivePrivilegeObjectType getObjectType() {
 (SEL_NOGRANT_AR, null));
     op2Priv.put(HiveOperationType.CREATETABLE_AS_SELECT, PrivRequirement.newPrivRequirementList(
         new PrivRequirement(SEL_NOGRANT_AR, IOType.INPUT),
+        new PrivRequirement(OWNER_INS_SEL_DEL_NOGRANT_AR, HivePrivilegeObjectType.DFS_URI),
         new PrivRequirement(OWNER_PRIV_AR, HivePrivilegeObjectType.DATABASE)));
 
     // QUERY,LOAD op can contain an insert & overwrite,
-- 
1.7.9.5

